### PR TITLE
Added PID in data from TPC, TOF for pi,K,P

### PR DESCRIPTION
**Pull request by rosarioturrisi**

### CODE REVIEW

File: DPG/Tasks/AOTTrack/qaMatchEff.cxx [hunk 1]
```diff
@@ -1108,56 +1183,115 @@ struct qaMatchEff {
...
+              histos.fill(HIST("data/ptptconfTPCo"), reco_pt, tpcinner_pt);
+          }
           histos.get<TH1>(HIST("data/qopthist_tpc"))->Fill(track.signed1Pt());
           histos.get<TH1>(HIST("data/pthist_tpc"))->Fill(trackPt);
           histos.get<TH1>(HIST("data/phihist_tpc"))->Fill(track.phi());
           histos.get<TH1>(HIST("data/etahist_tpc"))->Fill(track.eta());
+          //
+          // PID is applied
+          if (isPion) {
+            histos.get<TH1>(HIST("data/pthist_tpc_pi"))->Fill(ITStrackPt);
```
Comment (mfaggin) [inline DPG/Tasks/AOTTrack/qaMatchEff.cxx:1251]: here you fill with `ITStrackPt` but the histos are `tpc` ones. Fix it

File: DPG/Tasks/AOTTrack/qaMatchEff.cxx [hunk 2]
```diff
@@ -1108,56 +1183,115 @@ struct qaMatchEff {
...
-            histos.get<TH1>(HIST("data/qopthist_tpcits"))
-              ->Fill(track.signed1Pt());
+          } else { // DATA
+            if (makept2d)
+              histos.fill(HIST("data/ptptconfTPCITS"), reco_pt, tpcinner_pt);
+            histos.get<TH1>(HIST("data/qopthist_tpcits"))->Fill(track.signed1Pt());
+            //
+            //  PID is applied
+            if (isPion) {
+              histos.get<TH1>(HIST("data/pthist_tpcits_pi"))->Fill(ITStrackPt);
```
Comment (mfaggin) [inline DPG/Tasks/AOTTrack/qaMatchEff.cxx:1281]: here again: is `ITStrackPt` what you really want?

File: DPG/Tasks/AOTTrack/qaMatchEff.cxx [hunk 3]
```diff
@@ -1108,56 +1183,115 @@ struct qaMatchEff {
...
+              histos.get<TH1>(HIST("data/phihist_tpcits_pi"))->Fill(track.phi());
+              histos.get<TH1>(HIST("data/etahist_tpcits_pi"))->Fill(track.eta());
+            }
+            if (isKaon) {
+              histos.get<TH1>(HIST("data/pthist_tpcits_ka"))->Fill(ITStrackPt);
```
Comment (mfaggin) [inline DPG/Tasks/AOTTrack/qaMatchEff.cxx:1286]: here again: is ITStrackPt what you really want?

File: DPG/Tasks/AOTTrack/qaMatchEff.cxx [hunk 4]
```diff
@@ -1108,56 +1183,115 @@ struct qaMatchEff {
...
+              histos.get<TH1>(HIST("data/phihist_tpcits_ka"))->Fill(track.phi());
+              histos.get<TH1>(HIST("data/etahist_tpcits_ka"))->Fill(track.eta());
+            }
+            if (isProton) {
+              histos.get<TH1>(HIST("data/pthist_tpcits_pr"))->Fill(ITStrackPt);
```
Comment (mfaggin) [inline DPG/Tasks/AOTTrack/qaMatchEff.cxx:1291]: here again: is ITStrackPt what you really want?

File: DPG/Tasks/AOTTrack/qaMatchEff.cxx [hunk 5]
```diff
@@ -1108,56 +1183,115 @@ struct qaMatchEff {
...
           histos.get<TH1>(HIST("data/etahist_tpc"))->Fill(track.eta());
+          //
+          // PID is applied
+          if (isPion) {
+            histos.get<TH1>(HIST("data/pthist_tpc_pi"))->Fill(ITStrackPt);
+            histos.get<TH1>(HIST("data/phihist_tpc_pi"))->Fill(track.phi());
+            histos.get<TH1>(HIST("data/etahist_tpc_pi"))->Fill(track.eta());
+          }
+          if (isKaon) {
+            histos.get<TH1>(HIST("data/pthist_tpc_ka"))->Fill(ITStrackPt);
```
Comment (mfaggin) [inline DPG/Tasks/AOTTrack/qaMatchEff.cxx:1256]: here you fill with ITStrackPt but the histos are tpc ones. Fix it

File: DPG/Tasks/AOTTrack/qaMatchEff.cxx [hunk 6]
```diff
@@ -1108,56 +1183,115 @@ struct qaMatchEff {
...
+            histos.get<TH1>(HIST("data/phihist_tpc_ka"))->Fill(track.phi());
+            histos.get<TH1>(HIST("data/etahist_tpc_ka"))->Fill(track.eta());
+          }
+          if (isProton) {
+            histos.get<TH1>(HIST("data/pthist_tpc_pr"))->Fill(ITStrackPt);
```
Comment (mfaggin) [inline DPG/Tasks/AOTTrack/qaMatchEff.cxx:1261]: here you fill with ITStrackPt but the histos are tpc ones. Fix it

**Review by mfaggin — CHANGES_REQUESTED** [review]
Ciao @rosarioturrisi , thanks a lto for your implementations. I think that the `ITStrackPt` is making a bit of confusion... can you please check and in case fix it? See the detailed comments. Thank you.

**Comment by rosarioturrisi** [comment]
Thank you Mattia, indeed the ITStrackPt is there because maybe we want 
to see the difference

between the two definition of pt we use but it could not make sense in 
all cases, like f.e. if the track is ITS only. So we can choose

what to use where. As you notice I overlooked some editing doing 
copy-paste of lines, thanks for spotting now it's ok.

PR asap.

     R.

**Review by mfaggin — APPROVED** [review]

**Pull request merged**